include/ibamr/AllenCahnHierarchyIntegrator.h

```diff
@@ -0,0 +1,352 @@
+     * to calling initializePatchHierarchy().
+     */
+    void
+    initializeHierarchyIntegrator(SAMRAI::tbox::Pointer<SAMRAI::hier::PatchHierarchy<NDIM> > hierarchy,
+                                  SAMRAI::tbox::Pointer<SAMRAI::mesh::GriddingAlgorithm<NDIM> > gridding_alg) override;
+
+    /*!
+     * Prepare to advance the data from current_time to new_time.
+     */
+    void preprocessIntegrateHierarchy(double current_time, double new_time, int num_cycles = 1) override;
+
+    /*!
+     * Synchronously advance each level in the hierarchy over the given time
+     * increment.
+     */
+    void integrateHierarchy(double current_time, double new_time, int cycle_num = 0) override;
+
+    /*!
+     * Clean up data following call(s) to integrateHierarchy().
+     */
+    void postprocessIntegrateHierarchy(double current_time,
+                                       double new_time,
+                                       bool skip_synchronize_new_state_data,
+                                       int num_cycles = 1) override;
+
+    /*!
+     * \brief Register liquid fraction variable \f$ \varphi \f$.
+     */
+    void registerLiquidFractionVariable(SAMRAI::tbox::Pointer<SAMRAI::pdat::CellVariable<NDIM, double> > lf_var,
+                                        const bool output_lf_var = true) override;
+
+    /*!
+     * Reset cached hierarchy dependent data.
+     */
+    void
+    resetHierarchyConfigurationSpecialized(SAMRAI::tbox::Pointer<SAMRAI::hier::BasePatchHierarchy<NDIM> > hierarchy,
+                                           int coarsest_level,
+                                           int finest_level) override;
+
+    /*!
+     *  Compute the source terms of the energy equation.
+     */
+    void computeEnergyEquationSourceTerm(int F_scratch_idx, const double dt) override;
+
+    /*!
+     * Compute the source term for the Div U equation.
+     */
+    void computeDivergenceVelocitySourceTerm(int Div_U_F_idx, const double new_time) override;
+
+    /*!
+     * Set an object to provide boundary conditions for  \f$ \varphi \f$ variable,
+     * that has been registered with the hierarchy integrator.
+     */
+    void setLiquidFractionPhysicalBcCoef(SAMRAI::tbox::Pointer<SAMRAI::pdat::CellVariable<NDIM, double> > lf_var,
+                                         SAMRAI::solv::RobinBcCoefStrategy<NDIM>* lf_bc_coef);
+
+    /*!
+     * Return a \f$ \varphi \f$ boundary condition object.
+     */
+    SAMRAI::solv::RobinBcCoefStrategy<NDIM>* getLiquidFractionPhysicalBcCoef();
+
+    /*!
+     * Get the convective operator being used by this solver class for Allen-Cahn equation.
+     *
+     * If the convective operator has not already been constructed, then this
+     * function will initialize a default convective operator.
+     */
+    SAMRAI::tbox::Pointer<ConvectiveOperator>
+    getConvectiveOperatorForAllenCahnEquation(SAMRAI::tbox::Pointer<SAMRAI::pdat::CellVariable<NDIM, double> > lf_var,
```

Comment:
Similar to suggestions in a few other places:
```suggestion
    getAllenCahnEquationConvectiveOperator(SAMRAI::tbox::Pointer<SAMRAI::pdat::CellVariable<NDIM, double> > lf_var,
```

Reply:
👍